### PR TITLE
Integration of Alibaba Backend in OpenSDS

### PR DESCRIPTION
**Pull request by Click2Cloud-Gamma**

### CODE REVIEW

**Comment by kumarashit** [comment]
This doc appears to be very high-level. 
Ideally we should have:
a) Schema impacts
b) All APIs response/request (maybe existing but should be mentioned as this support will be using it)
c) Installation impact (N/A if not required)
d) UI impact

File: specs/daito/alibaba-provider.md [hunk 1]
```diff
@@ -0,0 +1,327 @@
...
+Using this feature, user will be able to do object management mechanism and migration for Alibaba Cloud in multi-cloud .
+### Motivation
+In existing OpenSDS Gelato project, we don't have any option using which user can manage object management mechanism and migration for Alibaba Cloud. So we adding  alibaba as backend for Object management and migration.
+### Goals
+
+* Add support of Alibaba OSS service in OpenSDS Gelato Project.
+
+
+
+### Design Details
```
Comment (kumarashit) [inline specs/daito/alibaba-provider.md]: This link is broken. Please check

**Comment by rhsakarpos** [comment]
hi
can we please split this into features, with details about 

- what needs to be done for Ali-baba backend, to support this feature
- any limitations, etc. 
- any changes in SODA API ?
- any changes in SODA schema ?
- any changes in any of our request/response ?
- functional differences between Alibaba and AWS (for example) and how this affects our functionality, if any

specifically for below features

1. add backend
2. create bucket
3. life-cycle rules
4. storage classes like hot, cold etc.
5. migration
6. replication
7. SSE

please add on if anything missed.

**Comment by Click2Cloud-Gamma** [comment]
> hi
> can we please split this into features, with details about
> 
> * what needs to be done for Ali-baba backend, to support this feature
> * any limitations, etc.
> * any changes in SODA API ?
> * any changes in SODA schema ?
> * any changes in any of our request/response ?
> * functional differences between Alibaba and AWS (for example) and how this affects our functionality, if any
> 
> specifically for below features
> 
> 1. add backend
> 2. create bucket
> 3. life-cycle rules
> 4. storage classes like hot, cold etc.
> 5. migration
> 6. replication
> 7. SSE
> 8. PUT and GET object
> 9. multipart upload
> 
> please add on if anything missed.

Thanks @rhsakarpos , we have modified required changes, please check. Thanks!

File: specs/daito/alibaba-provider.md (hunk 1 above)
Comment (kumarashit) [reply specs/daito/alibaba-provider.md]: @Click2Cloud-Gamma Can you please this part?

Comment (kumarashit) [reply specs/daito/alibaba-provider.md]: Clicking on the picture gives 404 error

**Comment by Click2Cloud-Gamma** [comment]
Please refer #52

**Comment by Click2Cloud-Gamma** [comment]
> This doc appears to be very high-level.
> Ideally we should have:
> a) Schema impacts
> b) All APIs response/request (maybe existing but should be mentioned as this support will be using it)
> c) Installation impact (N/A if not required)
> d) UI impact

@kumarashit Added, Please check on PR https://github.com/sodafoundation/design-specs/pull/52
Thanks.

File: specs/daito/alibaba-provider.md (hunk 1 above)
Comment (Click2Cloud-Gamma) [reply specs/daito/alibaba-provider.md:13]: This issue occured due to github functionality, after merging PR image can be viewed in file too.
Thanks

**Pull request closed without merging**